zeppelin-web/src/components/noteName-create/notename.controller.js

```diff
@@ -35,22 +36,23 @@ angular.module('zeppelinWebApp').controller('NotenameCtrl', function($scope, $ro
   };
 
   vm.preVisible = function(clone) {
-    var generatedName = vm.generateName();
-    $scope.note.notename = 'Note ' + generatedName;
+    $scope.note.notename = vm.generateName();
     vm.clone = clone;
     $scope.$apply();
   };
 
   vm.generateName = function () {
-    var DICTIONARY = [ '1', '2', '3', '4', '5', '6', '7', '8', '9', 'A', 'B',
-        'C', 'D', 'E', 'F', 'G', 'H', 'J', 'K', 'M', 'N', 'P', 'Q', 'R',
-        'S', 'T', 'U', 'V', 'W', 'X', 'Y', 'Z' ];
-    var randIndex, name = '';
-    for (var i = 0; i < 9; i++) {
-      randIndex = Math.floor(Math.random() * 32);
-      name += DICTIONARY[randIndex];
-    }
-    return name;
+    var newCount = 1;
+    angular.forEach(vm.notes.list, function (noteName) {
+      noteName = noteName.name;
+      if (noteName.match(/^Untitled Note [0-9]*$/)) {
+        var lastCount = noteName.substr(14) * 1;
+        if (newCount <= lastCount) {
```

Comment:
if we want to fill the holes (when you delete an element for example)
we should do:
```
if (newCount === lastCount) {
  newCount = lastCount + 1;
}
```
that way it would use the smaller available number